### PR TITLE
CLI to inspect lance dataset

### PR DESCRIPTION
**Pull request by eddyxu**
Writing in C++ to reuse lance code

### CODE REVIEW

File: cpp/src/lance/testing/io.cc [hunk 1]
```diff
@@ -28,6 +28,7 @@
 #include <string>
 
 #include "lance/arrow/file_lance.h"
+#include "lance/arrow/utils.h"
```
Comment (changhiskhan) [inline cpp/src/lance/testing/io.cc:31]: is this used in io.cc?

File: cpp/src/lance/arrow/file_lance.cc [hunk 2]
```diff
@@ -21,6 +21,7 @@
 #include <memory>
 
 #include "lance/arrow/file_lance_ext.h"
+#include "lance/format/format.pb.h"
```
Comment (changhiskhan) [inline cpp/src/lance/arrow/file_lance.cc:24]: where is this used?

File: cpp/src/lance/format/schema.cc [hunk 3]
```diff
@@ -702,4 +702,24 @@ pb::Field::Type Field::GetNodeType() const {
   }
 }
 
+void Print(const Field& field, const std::string& path, int indent = 0) {
```
Comment (changhiskhan) [inline cpp/src/lance/format/schema.cc:705]: would it make sense to print the storage type for extension types here as well?

**Review by changhiskhan — APPROVED** [review]
nice!

File: cpp/src/lance/format/schema.cc (hunk 3 above)
Comment (eddyxu) [reply cpp/src/lance/format/schema.cc:705]: Sure can do

File: cpp/src/lance/arrow/file_lance.cc (hunk 2 above)
Comment (eddyxu) [reply cpp/src/lance/arrow/file_lance.cc:24]: Removed. it was due to some forward declaration during dev.

File: cpp/src/lance/testing/io.cc (hunk 1 above)
Comment (eddyxu) [reply cpp/src/lance/testing/io.cc:31]: Fixed.

**Pull request merged**